src/ng/animate.js

```diff
@@ -416,8 +416,8 @@ var $AnimateProvider = ['$provide', /** @this */ function($provide) {
        * @param {Promise} animationPromise The animation promise that is returned when an animation is started.
        */
       cancel: function(runner) {
-        if (runner.end) {
-          runner.end();
+        if (runner.cancel) {
+          runner.cancel();
```

Comment:
I don't see why my fix would fail. It doesn't rely on the index but on the item itself (looking it up via `previousLeaveAnimations.indexOf(...)`). So, it is not affected by structural changes to the array.

Removing in reverse order should also work. (Why did I say I didn't like that?)